### PR TITLE
Enable ib-kubernetes deployment

### PR DESCRIPTION
**Pull request by almaslennikov**
ib-kubernetes provides a daemon that works in conjuction with [Mellanox InfiniBand SR-IOV CNI](https://github.com/Mellanox/ib-sriov-cni) and [Intel Multus CNI](https://github.com/intel/multus-cni), it acts on kubernetes Pod object changes(Create/Update/Delete), reading the Pod's network annotation and fetching its corresponding network CRD and and reads the PKey, to add the newly generated Guid or the predefined Guid in guid field of CRD cni-args to that PKey, for pods with annotation mellanox.infiniband.app.

This PR allows to deploy ib-kubernetes via network-operator with easy configuration.
This PR also contains some housekeeping work of enabling builds with go1.19+

### CODE REVIEW

File: deployment/network-operator/values.yaml [hunk 1]
```diff
@@ -215,6 +215,16 @@ sriovDevicePlugin:
     - name: hostdev
       vendors: [15b3]
 
+ibKubernetes:
+  deploy: false
+  image: ib-kubernetes
```
Comment (e0ne) [inline deployment/network-operator/values.yaml:220]: The image is not pullable:
```docker pull ghcr.io/mellanox/ib-kubernetes:v1.0.1
Error response from daemon: Head "https://ghcr.io/v2/mellanox/ib-kubernetes/manifests/v1.0.1": unauthorized```

File: manifests/stage-ib-kubernetes/0010-service_account.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2022 NVIDIA CORPORATION & AFFILIATES
```
Comment (e0ne) [inline manifests/stage-ib-kubernetes/0010-service_account.yaml:1]: Please, format licence headers like we do in the rest of files

File: manifests/stage-ib-kubernetes/0040-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: ib-kubernetes
+  namespace: {{ .RuntimeSpec.Namespace }}
+  annotations:
+    kubernetes.io/description: |
+      This deployment launches the ib-kubernetes daemon for Infiniband CNI.
```
Comment (e0ne) [inline manifests/stage-ib-kubernetes/0040-deployment.yaml:18]: What is `Infiniband CNI`?

File: manifests/stage-ib-kubernetes/0040-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          image: {{ .CrSpec.Repository }}/{{ .CrSpec.Image }}:{{ .CrSpec.Version }}
+          imagePullPolicy: IfNotPresent
+          command: ["/usr/bin/ib-kubernetes"]
+          resources:
+            requests:
+              cpu: 100m
+              memory: 300Mi
+          env:
+            - name: DAEMON_SM_PLUGIN
+              value: "ufm"
```
Comment (e0ne) [inline manifests/stage-ib-kubernetes/0040-deployment.yaml:82]: Should it be configurable via CRD?

File: manifests/stage-ib-kubernetes/0040-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            - name: DAEMON_PERIODIC_UPDATE
+              value: "5"
```
Comment (e0ne) [inline manifests/stage-ib-kubernetes/0040-deployment.yaml:84]: Is it 5m or 5s? Please, add comment.

Should it be configurable via CRD?

File: manifests/stage-ib-kubernetes/0040-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            - name: GUID_POOL_RANGE_START
+              value: {{ .CrSpec.PKeyGUIDPoolRangeStart }}
+            - name: GUID_POOL_RANGE_END
+              value: {{ .CrSpec.PKeyGUIDPoolRangeEnd }}
+            - name: UFM_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .CrSpec.UfmSecret }}
+                  key: UFM_USERNAME
+                  optional: true
```
Comment (e0ne) [inline manifests/stage-ib-kubernetes/0040-deployment.yaml:94]: According to this [1] code, username, password and address are mandatory fields

[1] https://github.com/Mellanox/ib-kubernetes/blob/803881ec421404170181a39f7ec255f82db8645f/pkg/sm/plugins/ufm/ufm.go#L45

**Review by e0ne — CHANGES_REQUESTED** [review]

File: manifests/stage-ib-kubernetes/0040-deployment.yaml (hunk 4 above)
Comment (almaslennikov) [reply manifests/stage-ib-kubernetes/0040-deployment.yaml:82]: There's only one plugin implemented at the moment, I don't think it needs to be configured at the moment.

File: deployment/network-operator/values.yaml (hunk 1 above)
Comment (almaslennikov) [reply deployment/network-operator/values.yaml:220]: The image has a private visibility on GHCR. I'm working with Mellanox org admins to change that

File: manifests/stage-ib-kubernetes/0040-deployment.yaml (hunk 3 above)
Comment (almaslennikov) [reply manifests/stage-ib-kubernetes/0040-deployment.yaml]: Changed to ib-sriov-cni

File: manifests/stage-ib-kubernetes/0040-deployment.yaml (hunk 4 above)
Comment (e0ne) [reply manifests/stage-ib-kubernetes/0040-deployment.yaml:85]: ok, sounds reasonable

File: deployment/network-operator/values.yaml (hunk 1 above)
Comment (almaslennikov) [reply deployment/network-operator/values.yaml:220]: Resolved. The image is publicly available now

**Review by e0ne — APPROVED** [review]

**Pull request merged**